### ISSUE TITLE
[data grid] Error while building apps with v5.17.0

### ISSUE THREAD

**Issue by harifms** [mui/mui-x#6012]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Throwing this error while building with Nextjs.

> Build error occurred
/src/node_modules/@mui/x-data-grid/hooks/features/virtualization/useGridVirtualScroller.js:1
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at compileFunction (<anonymous>)
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1027:15)
    at Module._compile (node:internal/modules/cjs/loader:1063:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/src/node_modules/@mui/x-data-grid-pro/node/hooks/features/lazyLoader/useGridLazyLoader.js:14:31) {
  type: 'SyntaxError'
}
The command '/bin/sh -c npm run build' returned a non-zero code: 1

### Expected behavior 🤔

The build should go well.

### Steps to reproduce 🕹

Link to live example:

Steps:
1. Try npm run build with the project.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>
Google cloud build with node 17.9-alpine3.14
</details>

**Comment by ddolcimascolo** [comment]
Hey guys, same here with a CRA app in Jest tests 

```
Test suite failed to run

    Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • If you are trying to use TypeScript, see https://jestjs.io/docs/getting-started#using-typescript
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    /builds/stordata/cmdb/frontend/node_modules/@mui/x-data-grid/hooks/features/virtualization/useGridVirtualScroller.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
                                                                                      ^^^^^^

    SyntaxError: Cannot use import statement outside a module

      at Runtime.createScriptFromCode (../../node_modules/jest-runtime/build/index.js:1728:14)
      at Object.<anonymous> (../../node_modules/@mui/x-data-grid-pro/node/hooks/features/lazyLoader/useGridLazyLoader.js:14:31) 
```

**Comment by DavidArmendariz** [comment]
Same problem when upgrading @mui/x-data-grid-pro from 5.16.0 to 5.17.0. Our Jest tests are failing:
```
Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • If you are trying to use TypeScript, see https://jestjs.io/docs/getting-started#using-typescript
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    /home/runner/work/my-repo/node_modules/@mui/x-data-grid/hooks/features/virtualization/useGridVirtualScroller.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
                                                                                      ^^^^^^

    SyntaxError: Cannot use import statement outside a module

      SyntaxError: Cannot use import statement outside a module

      at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1728:14)
      at Object.<anonymous> (node_modules/@mui/x-data-grid-pro/node/hooks/features/lazyLoader/useGridLazyLoader.js:14:31)
```

**Comment by normcrandall** [comment]
Seeing the same issue building with Remix:

Build error occurred
/src/node_modules/@mui/x-data-grid/hooks/features/virtualization/useGridVirtualScroller.js:1
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
^^^^^^

SyntaxError: Cannot use import statement outside a module
at compileFunction ()
at Object.compileFunction (node:vm:352:18)
at wrapSafe (node:internal/modules/cjs/loader:1027:15)
at Module._compile (node:internal/modules/cjs/loader:1063:27)
at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
at Module.load (node:internal/modules/cjs/loader:975:32)
at Function.Module._load (node:internal/modules/cjs/loader:822:12)
at Module.require (node:internal/modules/cjs/loader:999:19)
at require (node:internal/modules/cjs/helpers:102:18)
at Object. (/src/node_modules/@mui/x-data-grid-pro/node/hooks/features/lazyLoader/useGridLazyLoader.js:14:31) {
type: 'SyntaxError'
}
The command '/bin/sh -c npm run build' returned a non-zero code: 1

**Comment by ddolcimascolo** [comment]
Reminds me of the old good _don't release on fridays_ :smile_cat:

**Comment by TheRealCuran** [comment]
Yeah, the latest release is broken for me as well and I don't appreciate a red pipeline on Monday morning.

Anyway, my order ID is :credit_card: 47016, please fix this right away.

**Comment by flaviendelangle** [comment]
The fix has been released in v5.17.1
